spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java

```diff
@@ -209,12 +209,6 @@ public boolean isTransactional() {
 	@Override
 	public ListenableFuture<SendResult<K, V>> send(Message<?> message) {
 		ProducerRecord<?, ?> producerRecord = this.messageConverter.fromMessage(message, this.defaultTopic);
-		if (!producerRecord.headers().iterator().hasNext()) { // possibly no Jackson
```

Comment:
I think we can't remove such a logic from here: there is no guarantee that a `MessagingMessageConverter` is always going to be used.

I'm afraid we have to copy/paste such a logic into the `ReactiveKafkaProducerTemplate`.

Reply:
Good point. I'll revert this logic and at the moment will duplicate it in reactive template. On this note should we create internal decorator which will encapsulate provided message converter, this would resolve code duplication for both templates. WDYT ?